### ISSUE TITLE
Desktop docker for ubuntu is stoped

### ISSUE THREAD

**Issue by VladislavKutsobin** [docker/desktop-linux#19]
Hi

I have a problem with starting desktop docker fo ubuntu.
Version of Docker desktop 4.8.1
Docker version 20.10.16, build aa7e414

After start desktop app, sidebar on ubuntu pannel shows that "Docker desktop is stopped", in centre of app is "Docker desktop is stopped"
In left down side of aff docker image is gray and "Engine stopped"

What i do wrong?

Thank u

![dock1](https://user-images.githubusercontent.com/30027288/168493055-5a86e1fc-263b-48ec-9448-2259d74ba0d8.png)
![dock2](https://user-images.githubusercontent.com/30027288/168493057-e4698808-aba8-4653-ae2b-044cf6ee3114.png)
![dock3](https://user-images.githubusercontent.com/30027288/168493058-c10bc59c-4bda-4f00-95a7-5356c2ef7534.png)

**Comment by fclaudiopalmeira** [comment]
Same here

**Comment by smaccona** [comment]
@VladislavKutsobin @fclaudiopalmeira I had this issue as well. Are you running Ubuntu in a virtual machine, e.g. VirtualBox or similar? If so, you will need to enable nested virtualization (in VirtualBox, go to the virtual machine settings -> System tab -> Processor sub-tab -> check Enable Nested VT-x/AMD-V - other VM software should have a similar setting). This is because Docker Desktop creates a virtual machine for its Docker installation.

If you're not running Ubuntu in a virtual machine, then I'm not sure of the issue.

**Comment by fclaudiopalmeira** [comment]
@smaccona Indeed, I installed Ubuntu and tried that, it gave me the exact same error, and even though the nested Virtualization (Vmware) was enabled I had to install the virtualization environment:
`sudo apt -y install bridge-utils cpu-checker libvirt-clients libvirt-daemon qemu qemu-kvm`
Then I went back to my Centos Vm and it was exactly the same issue, installing the virtualization environment solved it:
`dnf install -y @virt virt-install`
If you need IOMMU (if you don't know what it is then you don't need it):
`grub2-editenv - set "$(grub2-editenv - list | grep kernelopts) intel_iommu=on"`
Reboot and you're all good to go.

## A little explanation of what is IOMMU:

> The IOMMU (I/O Memory Management Unit) is a system function that translates addresses used in DMA transactions, protects memory from illegal access by I/O devices, and remaps peripheral interrupts. It plays a critical role on IO virtualization technology which is widely used in today's mega-data center.

**Comment by p1-0tr** [comment]
@fclaudiopalmeira you shouldn't have to install `libvirt`, etc. to be able to run Docker Desktop. `qemu` should should get installed automatically since we declare it as a dependency. I suppose the VM installation wasn't configured to allow users access to `/dev/kvm`, and it's likely that one of the virt-* packages installs a udev config to allow that. You should be able to achieve the same by manually adding a udev rule to control access to `/dev/kvm` or adding your account to the `kvm` group (https://docs.docker.com/desktop/linux/install/#kvm-virtualization-support).

@VladislavKutsobin - as @smaccona suggested if you are running in a VM make sure you have nested virtualisation enabled. If that does not help, make sure your account is allowed to use KVM (https://docs.docker.com/desktop/linux/install/#kvm-virtualization-support). Failing that, could you upload diagnostics (https://docs.docker.com/desktop/linux/troubleshoot/#in-app-diagnostics) for us to be able to debug further?

**Comment by disdi** [comment]
Same here. All qemu and virtualization packages are present.

Pasting complete systemctl  logs of docker-desktop.service
[docker-desktop.log](https://github.com/docker/desktop-linux/files/8750375/docker-desktop.log)

**Comment by p1-0tr** [comment]
@disdi - did you make sure your user is allowed to use KVM (https://docs.docker.com/desktop/linux/install/#kvm-virtualization-support)? also could you paste the contents of your `~/.docker/desktop/log/host/com.docker.driver.amd64-linux.*.log`, or even better a full diagnostic (https://docs.docker.com/desktop/linux/troubleshoot/#in-app-diagnostics)?

**Comment by disdi** [comment]
Attaching logs related to virtualization  
[kvm.log](https://github.com/docker/desktop-linux/files/8751909/kvm.log)

Attaching logs related to docker-dekstop
[com.docker.driver.amd64-linux.stdout.log](https://github.com/docker/desktop-linux/files/8751916/com.docker.driver.amd64-linux.stdout.log)
[com.docker.driver.amd64-linux.stderr.log](https://github.com/docker/desktop-linux/files/8751918/com.docker.driver.amd64-linux.stderr.log)

Attaching logs of diagnostics tool
[20220523061123.zip](https://github.com/docker/desktop-linux/files/8751927/20220523061123.zip)

**Comment by p1-0tr** [comment]
@disdi - thanks. Based on the VM driver logs it looks like your `qemu` installation is missing (or can't access) a dependency:

    com.docker.driver.amd64-linux.stderr: qemu-system-x86_64: error while loading shared libraries: libxenctrl.so.4.11: cannot open shared object file: No such file or directory

I'd recommend looking into installing this and making sure you can manually run a VM (e.g. `qemu-system-x86_64 -accel kvm -cdrom <any distro install/livecd iso>`) on your system.

**Comment by disdi** [comment]
@p1-0tr Thanks for pointing this out.
There was a problem with my qemu package. Resolved it to fix the issue.

**Comment by gurpreet-fe** [comment]
Fixed this problem by enabling virtualisation from the BIOS settings.

**Comment by echeadle** [comment]
In disdi's docker-desktop log, two lines reproduced below, are the same messages I am getting.

When I checked .docker/desktop/http-proxy-control.sock  and /.docker/desktop/procd
Like the messages say, they do not exist.
What are they,  and do they stop Docker-Desktop from running?

Last error was Post "http://localhost/vm/httpproxy": dial unix <HOME>/.docker/desktop/http-proxy-control.sock: connect: no such file or directory

[com.docker.backend][W] ece62e23-PauseHDL /pause/events server not replying: Get "http://ipc/pause/events": dial unix <HOME>/.docker/desktop/procd.sock: connect: no such file or directory

**Comment by kusti420** [comment]
exact same problem on windows with IOMMU as well
